GVFS/GVFS.Common/GitHubUpgrader.cs

```diff
@@ -411,6 +465,74 @@ private static bool TryCreateDirectory(string path, out Exception exception)
             return true;
         }
 
+        private bool TryInstallGitUpgrade(GitVersion version, out string consoleError)
+        {
+            bool installSuccess = false;
+            using (ITracer activity = this.tracer.StartActivity(
+                $"{nameof(this.TryInstallGitUpgrade)}({version.ToString()})",
+                EventLevel.Informational))
+            {
+                if (!this.TryRunGitInstaller(out installSuccess, out consoleError) ||
+                    !installSuccess)
+                {
+                    EventMetadata metadata = new EventMetadata();
+                    metadata.Add("Upgrade Step", nameof(this.TryInstallGitUpgrade));
+                    this.tracer.RelatedError(metadata, $"{nameof(this.TryRunGitInstaller)} failed. {consoleError}");
+                    return false;
+                }
+
+                activity.RelatedInfo("Successfully installed Git version: " + version.ToString());
+            }
+
+            return installSuccess;
+        }
+
+        private bool TryInstallGVFSUpgrade(Version version, out string consoleError)
+        {
+            bool installSuccess = false;
+            using (ITracer activity = this.tracer.StartActivity(
+                $"{nameof(this.TryInstallGVFSUpgrade)}({version.ToString()})",
+                EventLevel.Informational))
+            {
+                if (!this.TryRunGVFSInstaller(out installSuccess, out consoleError) ||
+                !installSuccess)
+                {
+                    EventMetadata metadata = new EventMetadata();
+                    metadata.Add("Upgrade Step", nameof(this.TryInstallGVFSUpgrade));
+                    this.tracer.RelatedError(metadata, $"{nameof(this.TryRunGVFSInstaller)} failed. {consoleError}");
+                    return false;
+                }
+
+                activity.RelatedInfo("Successfully installed GVFS version: " + version.ToString());
+            }
+
+            return installSuccess;
+        }
+
+        private bool TryRunGitInstaller(out bool installationSucceeded, out string error)
+        {
+            error = null;
+            installationSucceeded = false;
+
+            int exitCode = 0;
+            bool launched = this.TryRunInstallerForAsset(GitAssetId, out exitCode, out error);
+            installationSucceeded = exitCode == 0;
+
+            return launched;
+        }
+
+        private bool TryRunGVFSInstaller(out bool installationSucceeded, out string error)
+        {
+            error = null;
+            installationSucceeded = false;
+
+            int exitCode = 0;
+            bool launched = this.TryRunInstallerForAsset(GVFSAssetId, out exitCode, out error);
+            installationSucceeded = exitCode == 0 || exitCode == RepoMountFailureExitCode;

```

Comment:
Aside from this line, the `TryRun*Installer` could be combined. Don't we run the VFS for Git installer with a parameter to not remount? Why do we need to special case that error code when the orchestrator would handle remounting?

Reply:
good point. I think this is a left over from initial days when installer itself used to do the re-mounting. Have removed it.